stream/street/import.js

```diff
@@ -17,78 +16,54 @@ function streamFactory(db, done){
   stats.tick();
 
   // create a new stream
-  return through.obj(function( batch, _, next ){
+  return through.obj(function (batch, _, next) {
 
-    // run serially so we can use transactions
-    db.serialize(function() {
+    // start transactio
+    db.transaction(() => {
 
-      // start transaction
-      db.run('BEGIN TRANSACTION', function(err){
+      // import batch
+      batch.forEach(function (street) {
 
-        // error checking
-        assert.transaction.start(err);
-
-        // import batch
-        batch.forEach( function( street ){
-
-          // insert names in to lookup table
-          street.getNames().forEach( function( name ){
-            stmt.names.run({
-              $id:   street.getId(),
-              $name: name
-            }, assert.statement.names);
+        // insert names in to lookup table
+        street.getNames().forEach(function (name) {
+          stmt.names.run({
+            id: street.getId(),
+            name: name
           });
-
-          // insert bbox in to rtree table
-          var bbox = street.getBbox();
-          stmt.rtree.run({
-            $id:   street.getId(),
-            $minX: bbox.minX,
-            $maxX: bbox.maxX,
-            $minY: bbox.minY,
-            $maxY: bbox.maxY
-          }, assert.statement.rtree);
-
-          // insert line in to polyline table
-          stmt.line.run({
-            $id:   street.getId(),
-            $line: street.getEncodedPolyline()
-          }, assert.statement.line);
-
         });
-      });
 
-      // commit transaction
-      db.run('END TRANSACTION', function(err){
-
-        // error checking
-        assert.transaction.end(err);
+        // insert bbox in to rtree table
+        var bbox = street.getBbox();
+        stmt.rtree.run({
+          id: street.getId(),
+          minX: bbox.minX,
+          maxX: bbox.maxX,
+          minY: bbox.minY,
+          maxY: bbox.maxY
+        });
 
-        // update statistics
-        stats.inc( batch.length );
+        // insert line in to polyline table
+        stmt.line.run({
+          id: street.getId(),
+          line: street.getEncodedPolyline()
+        });
 
-        // wait for transaction to complete before continuing
-        next();
       });
-    });
+    }).deferred();
 
-  }, function( next ){
+    // update statistics
+    stats.inc(batch.length);
 
-    // stop stats ticker
-    stats.tick( false );
+    // wait for transaction to complete before continuing
+    next();
 
-    // clean up
-    db.serialize(function(){
+  }, function (next) {
```

Comment:
this confused me for a while because the through2 docs don't make a lot of use of the cleanupfunction - so I think explicitly naming this function would be nice.

Also not calling the callback next but instead flushFunctionDone?